### PR TITLE
Remove -hexagon-small-data-threshold=0 hack (Issue #3739)

### PR DESCRIPTION
**Pull request by steven-johnson**
We believe this is no longer necessary for recent LLVM builds. https://github.com/halide/Halide/issues/3739

### CODE REVIEW

File: src/CodeGen_Hexagon.cpp [hunk 1]
```diff
@@ -57,37 +57,8 @@ CodeGen_Hexagon::CodeGen_Hexagon(Target t) : CodeGen_Posix(t) {
...
-    // TODO: This should be set on the module itself, or some other
-    // safer way to pass this through to the target specific lowering
-    // passes. We set the option here (after the base class'
-    // implementation of compile) because it is the last
-    // Hexagon-specific code to run prior to invoking the target
-    // specific lowering in LLVM, minimizing the chances of the wrong
-    // flag being set for the wrong module.
-    static std::once_flag set_options_once;
-    std::call_once(set_options_once, []() {
-        cl::ParseEnvironmentOptions("halide-hvx-be", "HALIDE_LLVM_ARGS",
```
Comment (dsharletg) [inline src/CodeGen_Hexagon.cpp]: This is unrelated to the small data threshold thing, and might still be necessary. I think @pranavb-ca's team was using this for something.

**Comment by dsharletg** [comment]
FWIW I think the reason this is different now is because we are loading shared objects now, and our linker can handle PIC code (and not because of a change in LLVM).

File: src/CodeGen_Hexagon.cpp (hunk 1 above)
Comment (steven-johnson) [reply src/CodeGen_Hexagon.cpp]: Hmm -- we can put this back in, but based on my experimentation in https://github.com/halide/Halide/pull/3734 this can be a dicey proposition, since we (theoretically) support using the same Halide instance from multiple threads; if you had a compilation setup that did some HVX and some non-HVX backends in the same process, you could potentially get different flag values depending on which backend happened to execute first. 

I'd definitely like to hear more about what their team is using it for.

Comment (pranavb-ca) [reply src/CodeGen_Hexagon.cpp]: I think we are fine to get rid of this. We haven't used this in our team for a long time now.

**Comment by steven-johnson** [comment]
@dsharletg are you ok with this?
